bpf-gpl/tc.c

```diff
@@ -904,19 +922,30 @@ static CALI_BPF_INLINE struct fwd calico_tc_skb_accepted(struct cali_tc_ctx *ctx
 		switch (ctx->ip_header->protocol) {
 		case IPPROTO_TCP:
 			tc_tcphdr(ctx)->source = bpf_htons(state->ct_result.nat_port);
+			if (state->ct_result.nat_sport) {
+				CALI_DEBUG("Fixing dest port from %d to %d\n",
```

Comment:
Perhaps "TCP" and "UDP" again here, and also something to indicate that it's a reverse SNAT as opposed to a forwards SNAT.